beacon-chain/sync/rpc_beacon_blocks_by_root.go

```diff
@@ -46,11 +47,40 @@ func (s *Service) sendBeaconBlocksRequest(ctx context.Context, requests *types.B
 		}
 		return nil
 	})
+
+	log := log.WithField("peer", id)
+
+	blkByRoot := make(map[[fieldparams.RootLength]byte]interfaces.ReadOnlySignedBeaconBlock)
 	for _, blk := range blks {
+		blkRoot, err := blk.Block().HashTreeRoot()
+		if err != nil {
+			return errors.Wrap(err, "block hash tree root")
+		}
+
+		blkByRoot[blkRoot] = blk
+	}
+
+	if err != nil {
```

Comment:
since this stuff is specific for debug maybe we should wrap the following so it doesn't loop through if it's not debugging?

// Check before logging
	if log.GetLevel() >= log.InfoDebug {
		// logic
	}

Reply:
Yeah, but in this case we should also modify this:
```
for _, blk := range blkByRoot {
```

because it would change with and without debug logs.

Do you see a nice solution?